kinesis-client-localstack/src/main/scala/kinesis4cats/client/producer/fs2/localstack/LocalstackFS2KinesisProducer.scala

```diff
@@ -100,15 +100,16 @@ object LocalstackFS2KinesisProducer {
         .withShardMapF(shardMapF)
         .build
       channel <- Channel
-        .bounded[F, (Record, Deferred[F, F[Unit]])](config.queueSize)
+        .bounded[
+          F,
+          (Record, Deferred[F, F[Producer.Result[PutRecordsResponse]]])
```

Comment:
**nitpick:** you can make this a `DeferredSink` rather than full `Deferred` here, making it obvious that the implementation is only allowed to push, not pull.